pkg/lib/semver/semver.go

```diff
@@ -0,0 +1,54 @@
+package semver
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+)
+
+// BuildIdCompare compares two versions and returns negative one if the first arg is less than the second arg, positive one if it is larger, and zero if they are equal.
+// This comparison follows typical semver precedence rules, with one addition: whenever two versions are equal with the exception of their build-ids, the build-ids are compared using prerelease precedence rules. Further, versions with no build-id are always less than versions with build-ids; e.g. 1.0.0 < 1.0.0+1.
+func BuildIdCompare(b semver.Version, v semver.Version) (int, error) {
```

Comment:
nit: would be nice to have a few tests directly confirming that this works as expected.